test/test_bot.py

```diff
@@ -593,7 +593,27 @@ def url_callback_http(bot, trigger, match):
 # -----------------------------------------------------------------------------
 # call_rule
 
-def test_call_rule(mockbot):
+@pytest.fixture
+def make_hello_event(mockbot):
```

Comment:
I was confused at first when reading the tests, because "make_hello_event" uses an overloaded word: "event". Since "make" is a generic verb, and "hello" refers to the text message, it wasn't clear exactly what it does. It could either generate an event (within an event system), or generate a trigger with an `HELLO` event.

Since it appears that this doesn't generate an `HELLO` trigger, but rather a `PRIVMSG` event with an `hello` text, and then match a rule based on that... I think this is a better name that represents what this fixture actually does:

```suggestion
def match_hello_rule(mockbot):
```

It match a hello rule, and now the return value of `re.match, Trigger, SopelWrapper` makes more sense to me.

And yes this means changing the 7 or more occurrences down the chain. 😁

Reply:
You might want to use the `triggerfactory`, maybe?

Reply:
Oh, and before I forget: a docstring would be nice. This is not as trivial as creating a single object, a bunch of stuff is done in here without explicit reasons at first. You have to read the tests below to understand the reason for this fixture, and since reading code can be difficult, the reason might remain unclear.

Reply:
Great suggestions all, implemented in c844758 and 4eea7a1